juju/juju
# Use model access

Update the model manager facade to use the access service for permissions rather than state. Part of the move to domains and dqlite.

Functionality of granting and revoking permissions on a model has been moved into the Access domain. Users() was not replaced in the modelmanager facade as it's being done as part of #17257 

Updated the model domain to give and remove permissions as models are created and deleted.

Secondarily, I found the dual uuids provided for creating a model to be highly error prone to the person just starting in this area. Updated for one uuid to be provided. All real calls provided a UUID, only the tests did it various ways. ModelType is only needed when calling the state version of create model, not the service version.

## Checklist


## QA steps

Ideally we could run `juju grant <username> write <modelname>` to test. However the checks on the other side, after you've logged in as a new user with permissions on a model that have been granted to you, aren't using the domain yet to verify. 

I've verified that UpdatePermission in the access domain is called. 

## Links

**Launchpad bug:** https://bugs.launchpad.net/juju/+bug/

**Jira card:** JUJU-5409
